### PR TITLE
KTOR-2563 Add GraalVM documentation

### PR DESCRIPTION
**Pull request by hhariri**

### CODE REVIEW

File: topics/graalvm.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+<microformat>
+<p>
+<control>Sample project</control>: <a href="https://github.com/ktorio/ktor-samples/tree/main/graalvm">graalvm</a>
+</p>
+</microformat>
+
+Ktor Server applications can make use of [GraalVM](https://graalvm.org) in order to have native images for different platforms, and of course,
+take advantage of the faster start-up times and other benefits that GraalVM provides. 
+
+Currently, Ktor Server applications that want to leverage GraalVM have to use CIO as the application engine.
```
Comment (andreyaksenov) [inline topics/graalvm.md:12]: please add a link to `Engines.md` for 'application engine'

File: topics/graalvm.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+</microformat>
+
+Ktor Server applications can make use of [GraalVM](https://graalvm.org) in order to have native images for different platforms, and of course,
+take advantage of the faster start-up times and other benefits that GraalVM provides. 
+
+Currently, Ktor Server applications that want to leverage GraalVM have to use CIO as the application engine. 
+
+## Preparing for GraalVM
+
+In addition to install GraalVM and having the installation directory in the system path, you need to prepare your application
```
Comment (andreyaksenov) [inline topics/graalvm.md:16]: `In addition to installing`? Or `In addition, to install ...`?

File: topics/graalvm.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+## Preparing for GraalVM
+
+In addition to install GraalVM and having the installation directory in the system path, you need to prepare your application 
+so that all dependencies are bundled, i.e. you need to create a [fatjar](fatjar.md).
+
+### Reflection Configuration 
+
+GraalVM has [some requirements](https://www.graalvm.org/reference-manual/native-image/Reflection/) when it comes to applications that use reflection, 
+which is the case of Ktor. It requires that you provide it a [JSON file](https://github.com/ktorio/ktor-samples/blob/main/graalvm/reflection.json) with 
+certain type information. This configuration file is then passed as an argument to to the `native-image` tool.
```
Comment (andreyaksenov) [inline topics/graalvm.md:23]: to to -> to

File: topics/graalvm.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+## Executing the `native-image` tool
+
+Once the fat jar is ready, the only step required is to create the native image, using the `native-image` tool. As this usually requires a
```
Comment (andreyaksenov) [inline topics/graalvm.md:27]: looks like the comma before 'using the `native-image` tool' is not required

File: topics/graalvm.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+GraalVM has [some requirements](https://www.graalvm.org/reference-manual/native-image/Reflection/) when it comes to applications that use reflection, 
+which is the case of Ktor. It requires that you provide it a [JSON file](https://github.com/ktorio/ktor-samples/blob/main/graalvm/reflection.json) with 
+certain type information. This configuration file is then passed as an argument to to the `native-image` tool.
+
+## Executing the `native-image` tool
+
+Once the fat jar is ready, the only step required is to create the native image, using the `native-image` tool. As this usually requires a 
+number of parameters, for convenience, the sample application provides a shell script ([build.sh](https://github.com/ktorio/ktor-samples/blob/main/graalvm/build.sh) for macOS/Linux 
+and [build.cmd](https://github.com/ktorio/ktor-samples/blob/main/graalvm/build.cmd) for Windows) that can be executed to obtain the final
+binary. You can use this script as starting point for your own applications. However, please note that some options may vary
```
Comment (andreyaksenov) [inline topics/graalvm.md:30]: as a starting point

File: topics/graalvm.md (hunk 1 above)
Comment (andreyaksenov) [inline topics/graalvm.md:12]: A [sample](https://github.com/ktorio/ktor-samples/tree/main/graalvm) has the following info about engines:
`Using the Netty engine is not compatible with GraalVM. Please following the corresponding issue for updates.`

Here we have `...have to use CIO as the application engine`. Am I correct that we support all except Netty? If yes, I'd wrote it here.

**Review by andreyaksenov — CHANGES_REQUESTED** [review]

File: topics/graalvm.md (hunk 1 above)
Comment (hhariri) [reply topics/graalvm.md:12]: No. We currently don't support jetty either.

**Review by andreyaksenov — APPROVED** [review]

**Pull request merged**